templates/restful_service/src/restful_service/main.bal

```diff
@@ -0,0 +1,150 @@
+
+            request.addHeader("SOAPAction", "urn:placeOrder");
+            request.setXmlPayload(<@untainted> soapEnv);
+            request.setHeader(mime:CONTENT_TYPE, mime:TEXT_XML);
+
+            var httpResponse = stockQuoteClient->post("/services/SimpleStockQuoteService", <@untainted> request);
+            if (httpResponse is http:Response) {
+                if (httpResponse.statusCode == 202) {
+                    httpResponse.statusCode = 201;
+                    httpResponse.reasonPhrase = "Created";
+                    httpResponse.setHeader("Location", "http://localhost:"
+                    + io:sprintf("%s", config:getAsInt("LISTENER_PORT")) + "/stockQuote/quote/" + symbol);
+
+                    // Create response message.
+                    xml responsePayload = xml `<ns:placeOrderResponse xmlns:ns="http://services.samples">
+                                                   <ns:response xmlns:ax21="http://services.samples/xsd"
+                                                   xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+                                                   xsi:type="ax21:placeOrderResponse">
+                                                       <ax21:status>Order has been created</ax21:status>
+                                                   </ns:response>
+                                               </ns:placeOrderResponse>`;
+                    httpResponse.setXmlPayload(<@untainted> responsePayload);
+                }
+                // Send response to the client.
+                self.respondToClient(caller, httpResponse, ERROR_MESSAGE_WHEN_RESPOND);
+            } else {
+                self.createAndSendErrorResponse(caller, <@untainted> <string>httpResponse.detail()?.message,
+                ERROR_MESSAGE_WHEN_RESPOND, 500);
+            }
+        } else {
+            self.createAndSendErrorResponse(caller, ERROR_MESSAGE_INVALID_PAYLOAD, ERROR_MESSAGE_WHEN_RESPOND, 400);
+        }
+    }
+
+    // Function to create the error response.
+    function createAndSendErrorResponse(http:Caller caller, string backendError, string errorMsg, int statusCode) {
+        http:Response response = new;
+        // Set status code to the error response.
+        response.statusCode = statusCode;
+        // Set the error message to the response payload.
+        response.setPayload(<string> backendError);
+        self.respondToClient(caller, response, errorMsg);
+    }
+
+    // Function to send the response back to the client.
+    function respondToClient(http:Caller caller, http:Response response, string errorMsg) {
+        // Send response to the caller.
+        var respond = caller->respond(response);
+        if (respond is error) {
+            log:printError(errorMsg, err = respond);
+        }
+    }
+
+    function constructSOAPPayload (xml payload, string namespace) returns xml {
```

Comment:
Shouldn't we use the SOAP connector? That way we can eliminate this envelope creation and setting action header.